File: loop-structures/Palindrome.java
import java.util.Scanner;
public class Palindrome{
    public static void main(String[] args){
    
    Scanner input = new Scanner(System.in);

    System.out.print("Enter a five digit integer: ");
    int digit = input.nextInt();

    while(true){
        if (digit >= 10000 && digit <= 99999){
            break;
        }
        else {
            System.out.print("Enter a five digit integer: ");
            digit = input.nextInt();
        }
    }
        int digitOne = (digit / 10000) % 10;
        int digitTwo = (digit / 1000) % 10;
        int digitThree = (digit / 100) % 10;
        int digitFour = (digit / 10) % 10;
        int digitFive = digit % 10;

        if (digitOne == digitFive && digitTwo == digitFour){
            System.out.println("It is a palindrome");
        }
        else {
            System.out.println("It is not a palindrome");
        }
    }
}

File: loop-structures/palindrome.py
digit = int(input("Enter a five digit integer: "))

while True:
    if digit >= 10000 and digit <= 99999:
        break;
    else:
        digit = int(input("Enter a five digit integer: "))

digit_one = (digit / 10000) % 10
digit_two = (digit / 1000) % 10
digit_three = (digit / 100) % 10
digit_four = (digit / 10) % 10
digit_five = digit % 10

if digit_one == digit_five and digit_two == digit_four:
    print("It is a palindrome")
else:
    print("It is not a palindrome")
